server.go

```diff
@@ -1525,6 +1527,14 @@ func (s *Server) serveConn(c net.Conn) error {
 			}
 		}
 
+		if !ctx.Request.isTLS {
+			if forwarded := ctx.Request.Header.PeekBytes(strForwarded); len(forwarded) > 0 {
+				ctx.Request.isTLS = bytes.Contains(forwarded, strProtoHTTPS)
+			} else {
+				ctx.Request.isTLS = bytes.Contains(ctx.Request.Header.PeekBytes(strXForwardedProto), strHTTPS)
+			}
+		}
```

Comment:
How about redirects from an https server to a http url? Shouldn't the check be both ways?